common/djangoapps/student/views.py

```diff
@@ -1576,6 +1590,8 @@ def create_account_with_params(request, params):
                 user.email_user(subject, message, from_address)
         except Exception:  # pylint: disable=broad-except
             log.error(u'Unable to send activation email to user from "%s"', from_address, exc_info=True)
+    else:
+        registration.activate()
```

Comment:
I'm happy to switch the other way. This was a small cleanup.

The effect is:

Case 1: The test cases still pass, so that's okay.
Case 2: ext_auth: It eliminates the need for a chunk of duplicate code here: 

https://github.com/edx/edx-platform/blob/59ad3ccb73a09f07947ab9be3c93dc328187ee89/common/djangoapps/external_auth/views.py#L217

I did not remove the duplicate code, since I thought that would necessitate more review, but it is a precursor for doing so. Aside from being redundant, the ext_auth code breaks abstraction barriers (directly setting is_active, rather than calling activate()). One possible catch: We would be removing the activation event in the audit logs, but (1) this event is a little non-sensical when activation is disabled for 3rd part auth and (2) we would not be losing any non-redundant information. 

Would you folks be okay giving it a :+1: if I were to just switch it back so behavior is as before? The cleanup isn't particularly a big deal.